training/tf/net2net.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import numpy as np
+import scipy.signal as signal
+from copy import deepcopy
+
+def convolve(w, x):
+    x_ch, x_w, x_h = x.shape
+    outputs, inputs, _, __ = w.shape
+    assert x_ch == inputs
+    res = np.zeros((outputs, x_w, x_h))
+    for o in range(outputs):
+        for c in range(inputs):
+            res[o,:,:] += signal.correlate2d(x[c,:,:], w[o,c,:,:], mode='same')
+    return res
+
+def read_net(filename):
+    with open(filename, 'r') as f:
+        weights = []
+        for e, line in enumerate(f):
+            if e == 0:
+                print("Version", line.strip())
+                if line != '1\n':
+                    raise ValueError("Unknown version {}".format(line.strip()))
+            else:
+                weights.append(list(map(float, line.split(' '))))
+            if e == 2:
+                channels = len(line.split(' '))
+                print("Channels", channels)
+        blocks = e - (4 + 14)
+        if blocks % 8 != 0:
+            raise ValueError("Inconsistent number of weights in the file")
+        blocks //= 8
+        print("Blocks", blocks)
+
+        return blocks, channels, weights
+
+def conv_bn_wider(weights, next_weights, inputs, channels,
+                  new_channels, noise_std=0, next_conv=True,
+                  verify=False):
+    """new_channels = number of channels to add"""
+
+    if new_channels == 0:
+        return weights, next_weights
+
+    rand = range(channels)
+    rand.extend(np.random.randint(0, channels, new_channels))
```

Comment:
In python 3, ranges cannot be directly extended:
```
AttributeError: 'range' object has no attribute 'extend'
```
Changing `rand = range(channels)` to `rand = list(range(channels))`, it works fine for me.